src/components/SingleArticle.jsx

```diff
@@ -1,33 +1,61 @@
-import { fetchArticleByID } from '../api-utils'
-import { useEffect, useState } from "react"
-import {useParams} from "react-router-dom"
+import { fetchArticleByID } from "../api-utils";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { patchVotes } from "../api-utils";
 
 const SingleArticle = () => {
-  const [currentArticle, setCurrentArticle] = useState({})
+  const [currentArticle, setCurrentArticle] = useState({});
+  const [isError, setIsError] = useState(false);
   const dateObj = new Date(currentArticle.created_at);
   const dateString = dateObj.toString();
-  const { article_id } = useParams()
-
+  const { article_id } = useParams();
+  const [userViewVotes, setUserViewVotes] = useState(0);
+  
 
   useEffect(() => {
     fetchArticleByID(article_id).then((article) => {
-      setCurrentArticle(article)
-    })
-  }, [article_id])
+      setCurrentArticle(article);
+    });
+  }, [article_id]);
 
+  const handleVoteClick = (increment) => {
+    setUserViewVotes(userViewVotes + increment);
+    patchVotes(currentArticle.article_id, increment).catch((err) => {
+      console.log(err);
+      setUserViewVotes(userViewVotes - increment);
```

Comment:
When using `setUserViewVotes` here you should pass a callback to get the current value of userViewVotes and not use it directly. With your current implementation you have 2 updates:

1. `setUserViewVotes(userViewVotes + increment)` -> set to 1 exactly
2. `setUserViewVotes(userViewVotes - increment)` -> set to -1 exactly

so when the catch triggers it'll go to -1 instead of back to 0.